### ISSUE TITLE
Encryption is broken

### ISSUE THREAD

**Issue by twitchyliquid64** [Matrix0xCC/gotunnel#3]
IV re-use in CTR mode ([here](https://github.com/Matrix0xCC/gotunnel/blob/master/src/secure/secure.go#L20)) permits a classic break for passive attackers (AKA complete compromise of confidentiality, at negligible cost).

Consider generating a random IV, and putting it at the start of your ciphertext output. You could even send it in your hello message (there is no need for IV to be secret).

Better still, use `crypto/tls`. The only annoying thing is generating the certificates and setting `InsecureSkipVerify` (im assuming your threat model is only against passive attackers here), but aside from that, this library will handle all the sneaky crypto for you. There should be tons of ppl who have done exactly this as well.

**Comment by Matrix0xCC** [comment]
@twitchyliquid64 You are right. TLS is a perfect solution, but currently simple encryption is enough for me.
Connections are created on demand now, any extra handshakes and verification will increase the latency. TLS may be supported after connection preparation is implemented.

**Comment by twitchyliquid64** [comment]
You might as well not have encryption at all, with your current setup plaintext can be recovered in a few thousand CPU cycles.

If you are concerned about latency, at least send a new IV in the client command and re-key that stream using it.

Still though, your best approach would be to have a few 'warm' connections open and ready, and just punt it down those with TLS.
